master/internal/api_checkpoint.go

```diff
@@ -73,17 +75,46 @@ func (m *Master) canDoActionOnCheckpoint(
 	return nil
 }
 
+func (m *Master) canDoActionOnCheckpointThroughModel(ctx context.Context, curUser model.User,
+	id uuid.UUID) error {
+	modelIDs, err := m.db.GetModelIDsAssociatedWithCheckpoint(id)
+	if err != nil {
+		return err
+	}
+	for _, id := range modelIDs {
+		model := &modelv1.Model{}
+		_ = m.db.QueryProto("get_model_by_id", model, id)
```

Comment:
feels suspicious the error result here is not checked. probably if err result is `db.ErrNotFound`, we need to return permission denied; err otherwise: https://github.com/determined-ai/determined/blob/master/master/internal/api_tasks.go#L73

Reply:
slacked you about this.